compiler/rustc_attr/src/builtin.rs

```diff
@@ -649,6 +629,30 @@ pub fn eval_condition(
 
                     !eval_condition(mis[0].meta_item().unwrap(), sess, features, eval)
                 }
+                sym::target => {
+                    if let Some(features) = features && !features.cfg_target_compact {
+                        feature_err(
+                            sess,
+                            sym::cfg_target_compact,
+                            cfg.span,
+                            &"compact `cfg(target(..))` is experimental and subject to change"
+                        ).emit();
+                    }
+
+                    mis.iter().fold(true, |res, mi| {
+                        res & {
+                            let mut mi = mi.meta_item().unwrap().clone();
+                            if mi.path.segments.len() == 1 {
```

Comment:
I don't think you even need this condition, multi-segment paths will be reported as an error on the next step anyway.
```rust
if let [seg, ..] = &mut mi.path.segments {
    seg.ident.name = Symbol::intern(&format!("target_{}", seg.ident.name))
}
```

Also style nit: all the code except for `eval_condition(...)` is better moved outside of the block.